service/controller/resource/tcnp/create.go

```diff
@@ -220,6 +220,9 @@ func (r *Resource) updateStack(ctx context.Context, cr infrastructurev1alpha2.AW
 		r.logger.LogCtx(ctx, "level", "debug", "message", "requested the update of the tenant cluster's node pool cloud formation stack")
 	}
 
+	// update node pool security group spec
+	cc.Spec.TenantCluster.TCNP.SecurityGroups = cc.Status.TenantCluster.TCNP.SecurityGroups
```

Comment:
uncertain if this is the right place where update the spec with up-to-date data

Reply:
This is wrong. The purpose of spec and status being separate is to express current and desired state. You would never set the content of a CR status field to some CR spec field. The reconciliation approach here is to render the SG template using the spec and the status tells us if we have to update the template, because when we have to update the status is behind, meaning status has fewer SGs than spec defines.

Reply:
I put it at the start so this is an update before the render, if this is still wrong, please give a bigger hint where it should be as I really have no idea and even when you mean it well, telling me where it should not be wont help me :)

Reply:
Ack, let's get together and pair a bit.